backend/core/src/main/java/org/sonarsource/sonarlint/core/repository/reporting/PreviouslyRaisedFindingsRepository.java

```diff
@@ -36,20 +40,30 @@ public class PreviouslyRaisedFindingsRepository {
   private final Map<String, Map<URI, List<RaisedIssueDto>>> previouslyRaisedIssuesByScopeId = new ConcurrentHashMap<>();
   private final Map<String, Map<URI, List<RaisedHotspotDto>>> previouslyRaisedHotspotsByScopeId = new ConcurrentHashMap<>();
 
-  public void addOrReplaceIssues(String scopeId, Map<URI, List<RaisedIssueDto>> raisedIssues) {
-    previouslyRaisedIssuesByScopeId.put(scopeId, raisedIssues);
+  public Map<URI, List<RaisedIssueDto>> addOrReplaceIssues(String scopeId, Map<URI, List<RaisedIssueDto>> raisedIssues) {
```

Comment:
Refactoring suggestion:

```suggestion

  public Map<URI, List<RaisedIssueDto>> addOrReplaceIssues(String scopeId, Map<URI, List<RaisedIssueDto>> raisedIssues) {
    return addOrReplaceFindings(scopeId, raisedIssues, previouslyRaisedIssuesByScopeId);
  }

  public Map<URI, List<RaisedHotspotDto>> addOrReplaceHotspots(String scopeId, Map<URI, List<RaisedHotspotDto>> raisedHotpots) {
    return addOrReplaceFindings(scopeId, raisedHotpots, previouslyRaisedHotspotsByScopeId);
  }

  private static  <F extends RaisedFindingDto> Map<URI, List<F>> addOrReplaceFindings(String scopeId, Map<URI, List<F>> raisedHotpots,
    Map<String, Map<URI, List<F>>> previouslyRaisedFindingsByScopeId) {
    var hotspotsPerFile = previouslyRaisedFindingsByScopeId.computeIfAbsent(scopeId, k -> new ConcurrentHashMap<>());
    hotspotsPerFile.putAll(raisedHotpots);
    return hotspotsPerFile;
  }
```

Reply:
Thanks, I added your suggestion